src/main/java/frc/trigon/robot/subsystems/climber/ClimberVisualization.java

```diff
@@ -0,0 +1,144 @@
+        this.targetFirstJointLigament = firstJointRoot.append(new MechanismLigament2d("TargetFirstJointLigament", ClimberConstants.DISTANCE_BETWEEN_JOINTS_METERS, ClimberConstants.MECHANISM_STARTING_ANGLE, ClimberConstants.MECHANISM_LINE_WIDTH, ClimberConstants.GRAY));
+        this.currentStringPositionLigament = stringRoot.append(new MechanismLigament2d("ZCurrentStringPositionLigament", ClimberConstants.CLOSED_STRING_LENGTH_METERS, ClimberConstants.CLOSED_STRING_ANGLE_DEGREES, ClimberConstants.MECHANISM_LINE_WIDTH, stringColor));
+        this.targetStringPositionLigament = stringRoot.append(new MechanismLigament2d("TargetStringPositionLigament", ClimberConstants.CLOSED_STRING_LENGTH_METERS, ClimberConstants.CLOSED_STRING_ANGLE_DEGREES, ClimberConstants.MECHANISM_LINE_WIDTH, ClimberConstants.GRAY));
+        currentStringPositionLigament.append(new MechanismLigament2d("CurrentStringConnectionLigament", ClimberConstants.STRING_CONNECTION_LIGAMENT_LENGTH, ClimberConstants.STRING_CONNECTION_LIGAMENT_ANGLE, ClimberConstants.MECHANISM_LINE_WIDTH, stringColor));
+        targetStringPositionLigament.append(new MechanismLigament2d("TargetStringConnectionLigament", ClimberConstants.STRING_CONNECTION_LIGAMENT_LENGTH, ClimberConstants.STRING_CONNECTION_LIGAMENT_ANGLE, ClimberConstants.MECHANISM_LINE_WIDTH, ClimberConstants.GRAY));
+
+        this.currentState = ClimberConstants.ClimberState.RESTING;
+    }
+
+    /**
+     * Updates the current climber arm angle and the string angle and the string length, then logs the mechanism.
+     *
+     * @param currentState    the current climber state used to determine the second joint pose
+     * @param currentPosition the current position of the climber
+     * @param targetPosition  the target position of the climber
+     */
+    public void update(ClimberConstants.ClimberState currentState, double currentPosition, double targetPosition) {
+        this.currentState = currentState;
+        Rotation2d currentPitch = getClimberFirstJointPitch(getStringLengthMeters(currentPosition));
+        Rotation2d targetPitch = getClimberFirstJointPitch(getStringLengthMeters(targetPosition));
+        currentFirstJointLigament.setAngle(ClimberConstants.MECHANISM_STARTING_ANGLE - currentPitch.getDegrees());
+        currentStringPositionLigament.setAngle(ClimberConstants.MECHANISM_STARTING_ANGLE - calculateStringAngle(currentPitch.minus(ClimberConstants.FIRST_JOINT_ANGLE_ADDITION), getStringLengthMeters(currentPosition)).getDegrees());
+        currentStringPositionLigament.setLength(getStringLengthMeters(currentPosition));
+        setTargetPosition(targetPosition, targetPitch);
+        logMechanism(currentPitch);
+    }
+
+    /**
+     * Logs the mechanism, and the poses.
+     *
+     * @param currentPitch the current pitch or the climber
+     */
+    public void logMechanism(Rotation2d currentPitch) {
```

Comment:
This doesn't only log the mechanism. Saying mechanism too much in this class will confuse people.